kubernetes-deployment/maintainlib/common.py

```diff
@@ -0,0 +1,176 @@
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import yaml
+import os
+import sys
+import subprocess
+import jinja2
+import argparse
+import paramiko
+
+
+
+def load_yaml_file(path):
+
+    with open(path, "r") as f:
+        file_data = yaml.load(f)
+
+    return file_data
+
+
+
+def execute_shell(shell_cmd, error_msg):
+
+    try:
+        subprocess.check_call( shell_cmd, shell=True )
+
+    except subprocess.CalledProcessError:
+        print error_msg
+        sys.exit(1)
+
+
+
+def read_template(template_path):
+
+    with open(template_path, "r") as fin:
+        template_data = fin.read()
+
+    return template_data
+
+
+
+def generate_from_template(template_data, cluster_config, host_config):
+
+    generated_file = jinja2.Template(template_data).render(
+        {
+            "hostcofig": host_config,
+            "clusterconfig": cluster_config['clusterinfo'],
+            "cluster": cluster_config
+        }
+    )
+
+    return generated_file
+
+
+
+def write_generated_file(generated_file, file_path):
+
+    with open(file_path, "w+") as fout:
+        fout.write(generated_file)
+
+
+
+def sftp_paramiko(src, dst, filename, host_config):
+
+    hostip = host_config['hostip']
+    username = host_config['username']
+    password = host_config['password']
+    port = 22
+    if 'sshport' in host_config:
+        port = host_config['sshport']
+
+    # First make sure the folder exist.
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh.connect(hostname=hostip, port=port, username=username, password=password)
+
+    stdin, stdout, stderr = ssh.exec_command("sudo mkdir -p {0}".format(dst), get_pty=True)
+    stdin.write(password + '\n')
+    stdin.flush()
+    for response_msg in stdout:
+        print response_msg.strip('\n')
+
+    ssh.close()
+
+    # Put the file to target Path.
+    transport = paramiko.Transport((hostip, port))
+    transport.connect(username=username, password=password)
+
+    sftp = paramiko.SFTPClient.from_transport(transport)
+    sftp.put('{0}/{1}'.format(src, filename), '{0}/{1}'.format(dst, filename))
+    sftp.close()
+
+    transport.close()
+
+
+
+def ssh_shell_paramiko(host_config, commandline):
```

Comment:
>ssh_shell_paramiko [](start = 4, length = 18)

all functions before this one have been used elsewhere (in service-deployment. can you refactor the code so k8s deployment and service deployment can reuse them?
Not necessary done in this PR though

Reply:
Better not. This will introduce dependency. Then we have to prepare another script for user to prepare environment,